src/main/java/com/mongodb/Read.java

```diff
@@ -0,0 +1,70 @@
+package com.mongodb;
+
+import com.mongodb.client.*;
+import com.mongodb.diagnostics.logging.Loggers;
+import org.bson.Document;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.mongodb.client.model.Filters.*;
+import static com.mongodb.client.model.Projections.*;
+import static com.mongodb.client.model.Sorts.descending;
+
+public class Read {
+
+    private static final Random rand = new Random();
+
+    public static void main(String[] args) {
+        Logger.getLogger(Loggers.PREFIX).setLevel(Level.WARNING);
+        try (MongoClient mongoClient = MongoClients.create(System.getProperty("mongodb.uri"))) {
+            MongoDatabase sampleTrainingDB = mongoClient.getDatabase("sample_training");
+            MongoCollection<Document> gradesCollection = sampleTrainingDB.getCollection("grades");
+
+            // find one document with new Document
+            Document student1 = gradesCollection.find(new Document("student_id", 10000)).first();
+            System.out.println("Student 1: " + student1.toJson());
+
+            // find one document with Filters.eq()
+            Document student2 = gradesCollection.find(eq("student_id", 10000)).first();
+            System.out.println("Student 2: " + student2.toJson());
+
+            // find a list of documents and iterate throw it using an iterator.
+            FindIterable<Document> iterable = gradesCollection.find(gte("student_id", 10000));
+            MongoCursor<Document> cursor = iterable.iterator();
+            System.out.println("Student list with a cursor: ");
+            while (cursor.hasNext()) {
+                System.out.println(cursor.next().toJson());
+            }
+
+            // find a list of documents and use a List object instead of an iterator
```

Comment:
Would using the `MongoIterable#foreach` helper be worth documenting - it saves the need to create a list of values.

Eg:

```
iterable.forEach((Consumer<Document>) document -> System.out.println(document.toJson()));
```

Reply:
I thought I did it line 53-54, no?

Reply:
Doh! please ignore